### PR TITLE
rewrite sysapi_query. new "smart" search.

### PR DESCRIPTION
**Pull request by UncleClapton**

### CODE REVIEW

**Review by theunkn0wn1 — CHANGES_REQUESTED** [review]
:hyperdilly:

File: ratlib/starsystem.py [hunk 1]
```diff
@@ -396,44 +396,44 @@ def refresh_bloom(bot, db=None):
...
-            response = requests.get('https://system.api.fuelrats.com/landmark?name={}'.format(system))
-            if response.status_code != 200:
-                return { "meta": { "error": "System API did not respond with valid data."} }
-            result = response.json()
-        except Timeout:
-            return { "meta": { "error": "The request to Systems API timed out!"} }
-        return result
+        queryurl += 'landmark?name={}'
+    elif querytype == "smart":
+        queryurl += 'mecha?name={}'
```
Comment (theunkn0wn1) [inline ratlib/starsystem.py]: string concatenation
image

File: ratlib/starsystem.py [hunk 2]
```diff
@@ -396,44 +396,44 @@ def refresh_bloom(bot, db=None):
...
-            result = response.json()['data']
-        except Timeout:
-            return { "meta": { "error": "The request to Systems API timed out!"} }
-        return result
+        queryurl += 'search?name={}'
+
+    try:
+        response = requests.get(queryurl.format(system))
+        if response.status_code != 200:
+            return { "meta": { "error": "System API did not respond with valid data." } }
```
Comment (theunkn0wn1) [inline ratlib/starsystem.py:413]: [question] wouldn't a non-2xx response be invalid?
shouldn't we actually /validate/ the returned object before making this assertion?

Comment (UncleClapton) [reply ratlib/starsystem.py:413]: not sure what you're asking here. Do you mean validate response?

Comment (theunkn0wn1) [reply ratlib/starsystem.py:413]:
```py
return { "meta": { "error": "System API did not respond with valid data." } }
```

**Review by theunkn0wn1 — APPROVED** [review]

**Comment by theunkn0wn1** [comment]
¯\_(ツ)_/¯

**Pull request merged**